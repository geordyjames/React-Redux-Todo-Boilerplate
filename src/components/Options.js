import React from 'react';
import Option from './Option';

const Options = props => (
	<div>
		<div className="widget-header">
			<h3 className="widget-header__title">Your Options</h3>
		</div>

		{props.options.length === 0 && (
			<p className="widget__message">Please add an option to get started!</p>
		)}
		{props.options.map((option, index) => (
			<Option
        key={option.id}
        id={option.id}
        optionText={option.name}
				count={index + 1}
				handleDeleteOption={props.handleDeleteOption}
			/>
		))}
	</div>
);

export default Options;
