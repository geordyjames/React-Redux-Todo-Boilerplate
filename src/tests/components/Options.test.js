import React from 'react';
import { shallow } from 'enzyme';
import todos from '../fixtures/todos';
import Options from '../../components/Options';

test('should render Options with empty todo  correctly', () => {
  const wrapper = shallow(<Options options={[]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Options with todos correctly', () => {
  const wrapper = shallow(<Options options={todos} handleDeleteOption={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});