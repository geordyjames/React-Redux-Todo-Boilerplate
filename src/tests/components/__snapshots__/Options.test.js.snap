// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Options with empty todo  correctly 1`] = `
<Fragment>
  <p
    className="uk-text-primary"
  >
    Please add an option to get started!
  </p>
  <ul
    className="uk-list uk-list-divider uk-padding"
  />
</Fragment>
`;

exports[`should render Options with todos correctly 1`] = `
<Fragment>
  <ul
    className="uk-list uk-list-divider uk-padding"
  >
    <Option
      count={1}
      handleDeleteOption={[Function]}
      id="1"
      key="1"
      optionText="Gum"
    />
    <Option
      count={2}
      handleDeleteOption={[Function]}
      id="2"
      key="2"
      optionText="Rent"
    />
    <Option
      count={3}
      handleDeleteOption={[Function]}
      id="3"
      key="3"
      optionText="Credit Card"
    />
  </ul>
</Fragment>
`;
